data-prepper-plugins/otel-trace-raw-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/oteltrace/OTelTraceRawProcessor.java

```diff
@@ -52,16 +56,23 @@ public class OTelTraceRawProcessor extends AbstractProcessor<Record<Span>, Recor
 
     private volatile boolean isShuttingDown = false;
 
-    public OTelTraceRawProcessor(final PluginSetting pluginSetting) {
-        super(pluginSetting);
-        traceFlushInterval = SEC_TO_MILLIS * pluginSetting.getLongOrDefault(
-                OtelTraceRawProcessorConfig.TRACE_FLUSH_INTERVAL, OtelTraceRawProcessorConfig.DEFAULT_TG_FLUSH_INTERVAL_SEC);
-        final int numProcessWorkers = pluginSetting.getNumberOfProcessWorkers();
+    @DataPrepperPluginConstructor
+    public OTelTraceRawProcessor(final OtelTraceRawProcessorConfig otelTraceRawProcessorConfig,
+                                 final PipelineDescription pipelineDescription,
+                                 final PluginMetrics pluginMetrics) {
+        super(pluginMetrics);
+        traceFlushInterval = SEC_TO_MILLIS * otelTraceRawProcessorConfig.getTraceFlushIntervalSeconds();
+        final int numProcessWorkers = pipelineDescription.getNumberOfProcessWorkers();
         traceIdTraceGroupCache = CacheBuilder.newBuilder()
                 .concurrencyLevel(numProcessWorkers)
-                .maximumSize(OtelTraceRawProcessorConfig.MAX_TRACE_ID_CACHE_SIZE)
-                .expireAfterWrite(OtelTraceRawProcessorConfig.DEFAULT_TRACE_ID_TTL_SEC, TimeUnit.SECONDS)
+                .maximumSize(otelTraceRawProcessorConfig.getTraceGroupCacheMaxSize())
+                .expireAfterWrite(otelTraceRawProcessorConfig.getTraceGroupCacheTimeToLive().toMillis(), TimeUnit.MILLISECONDS)
                 .build();
+
+        pluginMetrics.gauge(TRACE_GROUP_CACHE_COUNT_METRIC_NAME, traceIdTraceGroupCache, cache -> (double) cache.size());
```

Comment:
Shouldn't `traceIdTraceGroupCache` be long instead of casting to double?

Reply:
The current API expects a `double` I believe.